Bio/PDB/Structure.py

```diff
@@ -63,3 +94,85 @@ def internal_to_atom_coordinates(self, verbose: bool = False) -> None:
         """
         for chn in self.get_chains():
             chn.internal_to_atom_coordinates(verbose)
+
+    def _find_residue_by_id(self, chain: Chain, id_: int):
+        """
+        Return the residue with `id_` from the `chain`.
+
+        This is a workaround required due to Biopython's residue indexing e.g.
+        ('H_GLC', 100, 'A') instead of 100. For more info please refer to the
+        section "What is a residue id?" at
+        https://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ
+        """
+        for residue in chain.get_residues():
+            if residue.id[1] == id_:
+                return residue
+
+    def get_links(self) -> list[Link]:
+        """
+        Return intramolecular links of the Structure.
+
+        Each link consists of two Atom objects and their associated distance.
+        """
+        raw_links = self.header["links"]
+        links = []
+        for link in raw_links:
+            for model in self.get_models():
+                chain1 = model[link.res1.chain]
+                residue1 = self._find_residue_by_id(chain1, link.res1.id)
```

Comment:
You might want to have a guard here in case the structure is edited and the residue/chain/atom no longer exists.